src/components/CharacterKeyboardInput.cpp

```diff
@@ -35,6 +28,65 @@ namespace REGoth
     mMoveLeft    = bs::VirtualButton("Left");
     mMoveRight   = bs::VirtualButton("Right");
     mFastMove    = bs::VirtualButton("FastMove");
+    mAction      = bs::VirtualButton("Action");
+    mQuickSave   = bs::VirtualButton("QuickSave");
+
+    mCharacter = SO()->getComponent<Character>();
+
+    if (mCharacter.isDestroyed())
+    {
+      REGOTH_THROW(InvalidStateException,
+                   bs::StringUtil::format("Scene Object {0} does not have a Character component!",
+                                          SO()->getName()));
+    }
+
+    mCharacterAI = SO()->getComponent<CharacterAI>();
+
+    if (mCharacterAI.isDestroyed())
+    {
+      REGOTH_THROW(InvalidStateException,
+                   bs::StringUtil::format("Scene Object {0} does not have a CharacterAI component!",
+                                          SO()->getName()));
+    }
+
+    mEventQueue = SO()->getComponent<CharacterEventQueue>();
+
+    if (mEventQueue.isDestroyed())
+    {
+      REGOTH_THROW(
+          InvalidStateException,
+          bs::StringUtil::format("Scene Object {0} does not have a CharacterEventQueue component!",
+                                 SO()->getName()));
+    }
+  }
+
+  void CharacterKeyboardInput::update()
+  {
+    if (bs::gVirtualInput().isButtonDown(mAction))
```

Comment:
I think we need to think about how to handle input in non-continous cases.
This would trigger multiple times for one buttonpress (since it will be held down for more than one update cycle), no?

Reply:
No, this method is specifically for this use case. It will return true only for one single update. The `isButtonHeld` will do what you have in mind.

Reply:
True! 
I played around with this a little bit and I must have used fixedUpdate() or something, eventually moving to events eventhough just using update() would have fixed my problems :/.